r/R/dplyr-functions.R

```diff
@@ -442,3 +442,37 @@ nse_funcs$strptime <- function(x, format = "%Y-%m-%d %H:%M:%S", tz = NULL, unit
 
   Expression$create("strptime", x, options = list(format = format, unit = unit))
 }
+
+nse_funcs$wday <- function(x, label = FALSE, abbr = TRUE, week_start = getOption("lubridate.week.start", 7)) {
+  if (label) {
+    arrow_not_supported("Label argument")
+  }
+  offset <- get_date_offset(week_start)
+  Expression$create("add", Expression$create("day_of_week", x), Expression$scalar(offset))
+}
+
+#' Get date offset
+#' 
+#' Arrow's `day_of_week` kernel counts from 0 (Monday) to 6 (Sunday), whereas
+#' `lubridate::wday` counts from 1 to 7, and allows users to specify which day
+#' of the week is first (Sunday by default).  This function converts the returned
```

Comment:
I was going to mention that we actually also have an "ISO weekday" in the C++ kernels, but only as field in the `iso_calendar` kernel, and not as separate one. We could also add `iso_day_of_week` if that helps, or even just make `day_of_week` follow the ISO conventions. Because the 0 (monday) - 6 (sunday) might be something Python specific.

But then looked at what C++ does, and there the default is actually 0 (sunday) - 6 (saturday), so yet something else ;) (although I see that also Postgres uses that for `dow`)

In the end, once you have it in one form, it's an easy conversion to any of the other of course, so it might not matter that much.

Reply:
We could add options to the C++ kernel to enable different behaviors there.

Reply:
That probably makes more sense than my workaround here actually.

Reply:
If we go that way it would probably be best to have another Jira for "TemporalOptions". It's probably best you proceed with the workaround and we loop back to this later.

Reply:
OK, will do, and will open a JIRA, thanks!

Reply:
https://issues.apache.org/jira/browse/ARROW-13054

Reply:
Please leave a comment with that JIRA issue somewhere in this code that we expect to remove.

Reply:
Done